### ISSUE TITLE
Inner TRSM: SIMD build error; manifests in Ifpack2

### ISSUE THREAD

**Issue by mhoemmen** [kokkos/kokkos-kernels#183]
See Trilinos issue: https://github.com/trilinos/Trilinos/issues/2390

**Comment by nmhamster** [comment]
@kyungjoo-kim - you should be able to reproduce this error on Hansen.

**Comment by kyungjoo-kim** [comment]
@nmhamster 

How urgent is this ?

**Comment by kyungjoo-kim** [comment]
do you use cuda9 ? with serial host default space and cuda default exec space ?

**Comment by ndellingwood** [comment]
@kyungjoo-kim Like Christian mentioned in trilinos/Trilinos#2390, Cuda gets tested on White (Power8 + Kepler37) during integration testing, no x86. 
Previous testing of kokkos-kernels on kokkos-dev (SNB + Kepler35) using gcc/5.3 and cuda/8.0.44 Cuda_Serial and Cuda_OpenMP had all tests build and pass, in case this extra info is useful.

**Comment by nmhamster** [comment]
@kyungjoo-kim - using CUDA9.0 with GCC 4.9.3, `KOKKOS_ARCH=HSW;Kepler37`

**Comment by kyungjoo-kim** [comment]
@nmhamster 

I cannot reproduce the compiler error. Followings are my test environments.
```
KOKKOS_CXXFLAGS:STRING=--std=c++11;-expt-extended-lambda;-march=core-avx2;-mtune=core-avx2;-arch=sm_37
KOKKOS_CXXFLAGS:STRING=--std=c++11;-expt-extended-lambda;-march=core-avx2;-mtune=core-avx2;-arch=sm_37
[kyukim @hansen01] test > module list
Currently Loaded Modulefiles:
  1) autoconf/2.69.0                                            11) binutils/2.30.0                                            21) netcdf-fortran/4.4.4/openmpi/2.1.1/gcc/4.9.3/cuda/9.0.176
  2) automake/1.15.0                                            12) gcc/4.9.3                                                  22) metis/5.0.1/gcc/4.9.3
  3) libtool/2.4.6                                              13) cuda/9.0.176                                               23) parmetis/4.0.3/openmpi/2.1.1/gcc/4.9.3/cuda/9.0.176
  4) cmake/3.9.0                                                14) java/oracle/1.8.0                                          24) cgns/20171002/openmpi/2.1.1/gcc/4.9.3/cuda/9.0.176
  5) make/4.2.1                                                 15) numa/2.0.11                                                25) openblas/0.2.19/gcc/4.9.3
  6) m4/1.4.18                                                  16) openmpi/2.1.1/gcc/4.9.3/cuda/9.0.176                       26) superlu/4.3.0/gcc/4.9.3
  7) bison/3.0.4                                                17) pnetcdf-exo/1.8.1/openmpi/2.1.1/gcc/4.9.3/cuda/9.0.176     27) superlu-dist/4.3.0/openmpi/2.1.1/gcc/4.9.3/cuda/9.0.176
  8) flex/2.6.4                                                 18) zlib/1.2.11                                                28) boost/1.55.0
  9) git/2.9.4                                                  19) hdf5/1.10.1/openmpi/2.1.1/gcc/4.9.3/cuda/9.0.176           29) yamlcpp/0.5.3
 10) devtools/20170725                                          20) netcdf-exo/4.4.1.1/openmpi/2.1.1/gcc/4.9.3/cuda/9.0.176    30) devpack/openmpi/2.1.1/gcc/4.9.3/cuda/9.0.176

```

**Comment by nmhamster** [comment]
Did you run with the KOKKOS_ARCH set to HSW for Trilinos develop?

**Comment by kyungjoo-kim** [comment]
yes. when you have time, could we check if we use the same parameters ?

**Comment by nmhamster** [comment]
@kyungjoo-kim - are you around tomorrow (Friday)?

**Comment by mhoemmen** [comment]
@micahahoward may be interested in following this discussion.

**Comment by srajama1** [comment]
@kyungjoo-kim : Ping

**Comment by ndellingwood** [comment]
Should be resolved by #222 and #232
